icu4c/source/test/intltest/headeronlytest.cpp

```diff
@@ -0,0 +1,213 @@
+// © 2024 and later: Unicode, Inc. and others.
+// License & terms of use: http://www.unicode.org/copyright.html
+
+// headeronlytest.cpp
+// created: 2024dec11 Markus W. Scherer
+
+#include <string>
+#include <string_view>
+
+// Test header-only ICU C++ APIs. Do not use other ICU C++ APIs.
+#define U_SHOW_CPLUSPLUS_API 0
+#define U_SHOW_CPLUSPLUS_HEADER_API 1
+
+#include "unicode/utypes.h"
+#include "unicode/uset.h"
+#include "unicode/utf.h"
+#include "unicode/utf16.h"
+#include "intltest.h"
+
+using namespace std::string_view_literals;
```

Comment:
It might, in some cases, but while trying to make intltest work, I had a bit of back-and-forth with adding assertEquals() overloads and disambiguating call sites.

In my view, it's safe and harmless, and at most a bit overly specific.

Reply:
We really should just templatize assertEquals and make it work with heterogeneous types.
Not a blocker for this PR whose scope has already kept quite far obviously, but feel free to assign something to me to do that.

Reply:
Improving `assertEquals()` seems like a good idea but that's not something that's needed for this PR, just using `u""` without involving any string view literals works just fine and is simpler.

Reply:
I am removing the `sv` prefix now, and verified that it does work.
However, I don't understand what's *wrong* about using string_view literals.

Reply:
> I don't understand what's _wrong_ about using string_view literals.

It's an unnecessary complication that distracts from the real purpose of the code. You can, of course, always add additional code that does unnecessary but harmless things, but you normally don't because it makes your code harder to read.